### ISSUE TITLE
Python2: encoding declaration causes syntaxerror

### ISSUE THREAD

**Issue by novel-yet-trivial** [pgbovine/OnlinePythonTutor#194]
In python2 mode, this program:

    # -*- coding: utf-8 -*-
    print 'hello'

results in: 

    SyntaxError: encoding declaration in Unicode string (<string>, line 0)

**Comment by pgbovine** [comment]
thanks for report; unicode doesn't work great with python2, please try python3:
https://github.com/pgbovine/OnlinePythonTutor/blob/master/unsupported-features.md

**Comment by novel-yet-trivial** [comment]
Considering 17.4% of the people I help on your site are asking for help with python2 I think this is an issue you should address. Remember about half the python community is still stuck on python2, and last I looked about 90% of pypi downloads are for python2.

**Comment by pgbovine** [comment]
thanks for your note, and for helping out on the site. have you run into unicode-related issues when you've been tutoring? i've fixed a lot of them already; i'd like some more concrete test cases if possible

**Comment by pgbovine** [comment]
(aside from the test case you've already provide, of course!)

**Comment by novel-yet-trivial** [comment]
Yes, but to be honest they are pretty rare, since most python2 tutorials don't touch unicode. I'll add some more examples soon. However it would be nice at a minimum to allow the encoding declaration, since many good python2 examples include one as part of the standard header even though there is no unicode in the code.

**Comment by pgbovine** [comment]
thanks! will look into this when i get the chance; it's easily reproducible.

there's also this related issue, which i could also try to investigate at the same time:
https://github.com/pgbovine/OnlinePythonTutor/issues/124

**Comment by pgbovine** [comment]
FYI this might have something to do with it ...
https://stackoverflow.com/questions/39142304/syntaxerror-encoding-declaration-in-unicode-string

i'm investigating now ... have a hacky fix being prototyped but not certain that it won't break other stuff; may take me a while more to test on more examples. leaving some reference notes for myself for later:
- https://www.python.org/dev/peps/pep-0263/

**Comment by pgbovine** [comment]
@novel-yet-trivial -- whenever you have time, please let me know if you have additional examples of unicode causing problems with python 2. i'd like to have a few more test cases so that I can be more confident that my fixes won't accidentally cause other issues. thanks!
